x-pack/test/functional/apps/lens/smokescreen.ts

```diff
@@ -191,5 +109,76 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       expect(await PageObjects.lens.getLayerCount()).to.eql(2);
     });
+
+    it('should allow transition from line chart to donut chart and to bar chart', async () => {
+      await PageObjects.visualize.gotoVisualizationLandingPage();
+      await listingTable.searchForItemWithName('lnsXYvis');
+      await PageObjects.lens.clickVisualizeListItemTitle('lnsXYvis');
+      await PageObjects.lens.goToTimeRange();
+      await PageObjects.lens.switchToVisualization('donut');
+
+      expect(await PageObjects.lens.getTitle()).to.eql('lnsXYvis');
+      expect(await PageObjects.lens.getDimensionTriggerText('lnsPie_sliceByDimensionPanel')).to.eql(
+        'Top values of ip'
+      );
+      expect(await PageObjects.lens.getDimensionTriggerText('lnsPie_sizeByDimensionPanel')).to.eql(
+        'Average of bytes'
+      );
+
+      await PageObjects.lens.switchToVisualization('bar');
+      expect(await PageObjects.lens.getTitle()).to.eql('lnsXYvis');
```

Comment:
We could check whether the dimensions are retained when switching to bar, or is this handled somewhere else already?

Reply:
Sure, I'll add it!